### PR TITLE
Added fixes and refactoring for the next test case class

### PR DESCRIPTION
**Pull request by luishernandezmonge**
### Ticket reference or full description of what's in the PR
-Crafter3LoadTest1Script.java
Also resolved some issues with contextclick method on WebDriverManager

### CODE REVIEW

File: src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java [hunk 1]
```diff
@@ -203,7 +200,7 @@ public void prepareTestArea() {
 
 		dashboardPage.rightClickCopyContentPage(styleCategoryLandingStyle);
 
-		this.driverManager.isElementPresentAndClickableByXpath(bigTree1FolderLocator);
+		// this.driverManager.isElementPresentAndClickableByXpath(bigTree1FolderLocator);
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java]: Please remove commented out code (here and elsewhere in this PR).

File: src/test/java/org/craftercms/studio/test/utils/WebDriverManager.java [hunk 2]
```diff
@@ -46,44 +49,49 @@ public void openConnection() {
...
+					capabilities = DesiredCapabilities.internetExplorer();
+					capabilities.setCapability(InternetExplorerDriver.INTRODUCE_FLAKINESS_BY_IGNORING_SECURITY_DOMAINS,
+							true);
+					InternetExplorerOptions internetExplorerOptions = new InternetExplorerOptions(capabilities);
+
+					System.setProperty("webdriver.ie.driver", envProperties.getProperty("ie.driver.path"));
+					driver = new InternetExplorerDriver(internetExplorerOptions);
+					break;
+				case "chrome":
+//				    capabilities = DesiredCapabilities.chrome();
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/utils/WebDriverManager.java]: Please remove commented out code, here and everywhere else in this PR.

File: src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java [hunk 3]
```diff
@@ -286,23 +280,25 @@ public void editSelectedContent() {
...
+			this.driverManager
+					.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath", historySecondItemCheckbBox)
+					.click();
 
 			// click on Compare button
 			dashboardPage.clickCompareButton();
 
 			// switching to the compare frame
-			//driverManager.getDriver().switchTo().frame(differencesDialogId);
+			// driverManager.getDriver().switchTo().frame(differencesDialogId);
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java]: Please remove commented out code.

File: src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java [hunk 4]
```diff
@@ -605,52 +576,42 @@ public void step11() {
...
-		WebElement anotherTestBigTreeChildFolder = this.driverManager
-				.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath", anotherTestBigTreeChildFolderLocator);
+		WebElement anotherTestBigTreeChildFolder = this.driverManager.waitUntilElementIsClickable("xpath",
+				anotherTestBigTreeChildFolderLocator);
 		dashboardPage.expandParentFolder(anotherTestBigTreeChildFolder);
 
-		this.driverManager.isElementPresentAndClickableByXpath(anotherTestBigTreeChildFolderLocator);
-		this.driverManager.isElementPresentAndClickableByXpath(
+		this.driverManager.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath",
 				anotherTestBigTreeChildFolderLocator + "/../../../../../div/div[5]/table/tbody/tr/td/span");
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java]: Is there a better way to do this? (other than this xpath)

File: src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java [hunk 5]
```diff
@@ -605,52 +576,42 @@ public void step11() {
...
 
-		this.driverManager.isElementPresentAndClickableByXpath(bigTree2BigTree1ChildFolderLocator);
 		bigTree2BigTree1ChildFolder = this.driverManager
 				.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath", bigTree2BigTree1ChildFolderLocator);
 		dashboardPage.expandParentFolder(bigTree2BigTree1ChildFolder);
 
-		this.driverManager.isElementPresentAndClickableByXpath(bigTree2BigTree1ChildFolderLocator);
-		this.driverManager.isElementPresentAndClickableByXpath(
+		this.driverManager.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath",
 				bigTree2BigTree1ChildFolderLocator + "/../../../../../div/div[5]/table/tbody/tr/td/span");
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java]: Is there a better way to do this?

File: src/test/java/org/craftercms/studio/test/pages/DashboardPage.java [hunk 6]
```diff
@@ -209,8 +209,10 @@ public DashboardPage(WebDriver driver) {
 	public void clickPagesTree() {
 		this.driverManager.waitUntilSidebarOpens();
 		WebElement expandPagesTree = this.driverManager.driverWaitUntilElementIsPresentAndDisplayed("xpath", pagesTree);
+		if(!expandPagesTree.getAttribute("class").contains("open")) {
 		expandPagesTree.click();
 		driverManager.waitUntilFolderOpens("xpath", ".//a[@id='pages-tree']");
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/pages/DashboardPage.java]: Is there a better way to do this?

File: src/test/java/org/craftercms/studio/test/pages/DashboardPage.java [hunk 7]
```diff
@@ -251,20 +253,19 @@ public void clickHomeTree() {
...
-            WebElement addContent = this.driverManager.driverWaitUntilElementIsPresentAndDisplayed("xpath",
-                    addNewContent);
-            addContent.click();
-        });
-    }
-
+	public void rightClickHome() {
+		this.driverManager.waitUntilPageLoad();
+		this.driverManager.waitUntilSidebarOpens();
+		this.driverManager.waitUntilFolderOpens("xpath", ".//a[@id='pages-tree']");
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/pages/DashboardPage.java]: Is there a better way to do this?

File: src/test/java/org/craftercms/studio/test/pages/DashboardPage.java [hunk 8]
```diff
@@ -283,19 +284,18 @@ public void rightClickSpecificFolder(String folderLocation) {
...
 	// Press right click select new folder
 	public void rightClickNewFolderOnHome() {
 		// wait for the animation to end
-		driverManager.waitUntilSidebarOpens();
-
-		WebElement home = this.driverManager.driverWaitUntilElementIsPresentAndDisplayedAndClickable("xpath",
-				homeContent);
-		this.getDriverManager().contextClick(this.getDriverManager().getDriver(), home, false);
+		this.driverManager.waitUntilSidebarOpens();
+		this.driverManager.waitUntilFolderOpens("xpath", ".//a[@id='pages-tree']");
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/test/pages/DashboardPage.java]: Is there a better way to do this?

File: src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java (hunk 4 above)
Comment (luishernandezmonge) [reply src/test/java/org/craftercms/studio/test/cases/generaltestcases/Crafter3LoadTest1Script.java]: For xpaths like: folderLocator + "/../../../../../div/div[1]/table/tbody/tr/td/span" this is a dynamical xpath that depends on the trees structure created during test case execution and it depends totally on the folder (name&location) that contains the element to be manipulated.

**Comment by luishernandezmonge** [comment]
For xpaths like: folderLocator + "/../../../../../div/div[1]/table/tbody/tr/td/span" this is a dynamical xpath that depends on the trees structure created during test case execution and it depends totally on the folder (name&location) that contains the element to be manipulated.

**Pull request merged**